test/rules/prefer_is_empty.dart

```diff
@@ -4,17 +4,33 @@
 
 // test w/ `pub run test -N prefer_is_empty`
 
+const l = '';
+const bool empty = l.length == 0; //OK
+
+class A {
+  final List<String> a;
+  const A(this.a) : assert(a.length > 0); //OK
```

Comment:
It would be good to have some tests using a non-const constructor to show that we still generate the lint in those cases.

Reply:
Yes.  Good idea.